src/library/scala/beans/ScalaBeanInfo.scala

```diff
@@ -32,15 +31,9 @@ abstract class ScalaBeanInfo(clazz: java.lang.Class[_],
   override def getPropertyDescriptors() = pd
   override def getMethodDescriptors() = md
 
-  // override def getAdditionalBeanInfo() = Array(Introspector getBeanInfo clazz.getSuperclass)
-
   private def init() {
-    var i = 0
```

Comment:
Indeed - having a guarantee that range foreach is always transformed would be beautiful.
I thought Miguel also did some work there a few years back.